### PR TITLE
Update WGSL tests

### PR DESCRIPTION
**Pull request by sarahM0**
This CL:
Cleans up WGSL validation tests to follow entry point spec changes.
Adds cts tests for switch statements.
Fixes other minor syntax issues, eg. adding parenthesis for location, offset, ...

### CODE REVIEW

File: src/webgpu/shader/validation/wgsl/duplicate-var-name-within-func.fail.wgsl [hunk 1]
```diff
@@ -1,13 +1,11 @@
-# v-0014 - This fails because of the duplicated `a` variable.
+# v-0014 - This passes because variable `a` is redeclared.
```
Comment (kainino0x) [inline src/webgpu/shader/validation/wgsl/duplicate-var-name-within-func.fail.wgsl:1]: I know we agreed on lexical scoping, but did we agree to allow shadowing? @dj2

File: src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause.fail.wgsl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# v-0008: line 6: switch statement must have exactly one default clause
```
Comment (kainino0x) [inline src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause.fail.wgsl:1]: I keep losing track of this one - is this what we agreed? I thought it was okay to not have a default clause (the semantics should be pretty obvious I think). Not trying to change this, just want to make sure it's the right thing.

**Review by kainino0x — APPROVED** [review]

File: src/webgpu/shader/validation/wgsl/duplicate-entry-point.fail.wgsl [hunk 3]
```diff
@@ -1,12 +1,11 @@
 # v-0020 - Duplicate entry point <name, stage> pair.
```
Comment (dj2) [inline src/webgpu/shader/validation/wgsl/duplicate-entry-point.fail.wgsl:1]: This isn't quite true any more. You can't have a duplicate entry point name at all anymore, regardless of the stage. So, we should add a second test with `stage(vertex)` and `stage(fragment)` with the same name to verify it fails.

File: src/webgpu/shader/validation/wgsl/duplicate-var-name-within-func.fail.wgsl (hunk 1 above)
Comment (dj2) [reply src/webgpu/shader/validation/wgsl/duplicate-var-name-within-func.fail.wgsl:1]: As far as I remember, shadowing is _not_ allowed.

File: src/webgpu/shader/validation/wgsl/switch-case-selector-must-have-the-same-type-as-the-selector-expression-2.fail.wgsl [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# v-0026: line 7: the case selector values must have the same type as the selector expression
+
+[[stage(vertex)]]
+fn main() -> void {
+  var a: i32 = -2;
```
Comment (dj2) [inline src/webgpu/shader/validation/wgsl/switch-case-selector-must-have-the-same-type-as-the-selector-expression-2.fail.wgsl:5]: Can we add a passing case for switch statements?

File: src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause-2.fail.wgsl [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# v-0008: line 6: switch statement must have exactly one default clause
```
Comment (dj2) [inline src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause-2.fail.wgsl:1]: nit: s/exactly one/at most one/

File: src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause.fail.wgsl (hunk 2 above)
Comment (dj2) [reply src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause.fail.wgsl:1]: I believe it's `at most one` although the spec isn't updated yet. @dneto0 to verify my memory.

File: src/webgpu/shader/validation/wgsl/duplicate-var-name-within-func.fail.wgsl (hunk 1 above)
Comment (kainino0x) [reply src/webgpu/shader/validation/wgsl/duplicate-var-name-within-func.fail.wgsl:1]: sg, in that case we should test both the old test (siblings, passes) and new test (nested, fails)

File: src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause-2.fail.wgsl (hunk 5 above)
Comment (dj2) [reply src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause-2.fail.wgsl:1]: I'm wrong, @dneto0 pointed out must have is correct.

File: src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause.fail.wgsl (hunk 2 above)
Comment (dj2) [reply src/webgpu/shader/validation/wgsl/switch-must-have-exactly-one-default-clause.fail.wgsl:1]: I'm wrong, it's must have currently.

**Comment by sarahM0** [comment]
Sorry for the delay, I totally forgot about this PR.

**Review by dj2 — APPROVED** [review]

**Pull request merged**